ffmpeg/api_test.go

```diff
@@ -1212,7 +1212,9 @@ func audioOnlySegment(t *testing.T, accel Acceleration) {
 			Accel:   accel,
 		}}
 		_, err := tc.Transcode(in, out)
-		if err != nil {
+		if i == 2 && (err == nil || err.Error() != "No keyframes in input") {
+			t.Error("Expected to fail for audio-only segment midstream but did not")
```

Comment:
Just to clarify - at the moment, the transcoder is able to handle audio-only segments mid-stream (we currently only reject audio-only segments at the beginning of a stream), but after this PR the transcoder will no longer be able to handle audio-only segments because they don't have any keyframes? 

Should the transcoder instead continue accepting audio-only segments mid-stream, but only error out on > 0 frame segments with no keyframes? Perhaps by returning `lpms_ERR_INPUT_NOKF` from the cgo code only if no keyframes were detected *and* at least 1 frame was fed to the decoder?

Reply:
Yep this is an intentional stop-gap decision to treat both the 0-video-packets and 0-keyframe-video-packets cases the same way.

A con of this is, as you mention, we no longer support 0-video-packets segments mid-stream even though they could be supported. IIRC my testing of that particular usecase anyway did not work well with some end-user HLS players as their player's "decoder" also had similar issues around non-existent video metadata midstream. So if we don't send that segment back to the user's player altogether, things might actually be smoother.

A pro of this stop-gap is that we don't need to separately create a new testcase for both, and it keeps the underlying code clean and doesn't introduce a new "frames decoded" counter, which we can reconsider and add later on.

Reply:
> IIRC my testing of that particular usecase anyway did not work well with some end-user HLS players as their player's "decoder" also had similar issues around non-existent video metadata midstream.

I thought this was just for audio-only segments at the beginning of a stream which is why the transcoder will return an error immediately for audio-only segments at the beginning of a stream? Reference #207.

If the transcoder does handle audio-only segments mid-stream smoothly right now, I think we should avoid changing that behavior here if possible.

Reply:
Done, also turns out I already had a "frames sent" counter called `pkt_diff` from earlier, so was able to re-use it ezpz :)